### PR TITLE
Adds support for Multiple products

### PR DESCRIPTION
**Pull request by nullbear**
Adds support for multiple recipe results.
Makes it so that recipes can have multiple biproducts, instead of just
one. The results list works just like the reagents list.

ex. list("reagentid" = 5, "otherreagent" = 3) where 5 and 3 are the
amounts produced.

ALSO ADDED SUPPORT FOR COLD REACTIONS. IE. cooling a mix to react, instead of heating it.

recipe on_reaction 'created volume' is the multiplier for the recipe,
not the sum of the volume of the biproducts. Equations have been
adjusted to this change.

A couple of reactions are made slightly more efficient.

Removes a couple of trailing returns and unneccessary variable
declarations.

This is a refactor.

### CODE REVIEW

**Comment by PKPenguin321** [comment]
I thought this would be about ~~table~~crafting but this is really good too

**Comment by nullbear** [comment]
Tablecrafting would b hard though cause it uses a reference of the product and tries to put stuff in it.

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -382,11 +382,10 @@ var/const/INJECT = 5 //injection
 					for(var/B in C.required_reagents)
 						remove_reagent(B, (multiplier * C.required_reagents[B]), safety = 1)
 
-					var/created_volume = C.result_amount*multiplier
-					if(C.result)
-						feedback_add_details("chemical_reaction","[C.result]|[C.result_amount*multiplier]")
+					for(var/result in C.results)
```
Comment (phil235) [inline code/modules/reagents/chemistry/holder.dm]: no a great var name, easy to get confused.

File: code/modules/reagents/chemistry/recipes/others.dm [hunk 2]
```diff
@@ -2,211 +2,183 @@
...
-	result = "sterilizine"
+	results = list("sterilizine" = 3)
 	required_reagents = list("ethanol" = 1, "charcoal" = 1, "chlorine" = 1)
-	result_amount = 3
 
 /datum/chemical_reaction/lube
 	name = "Space Lube"
 	id = "lube"
-	result = "lube"
+	results = list("lube" = 3) // fuck this spillover into nullspace.
```
Comment (phil235) [inline code/modules/reagents/chemistry/recipes/others.dm]: These comments are not needed, they don't say anything about the code.

**Comment by nullbear** [comment]
Oh dear i added an infinite loop somewhere.

**Comment by Jalleo** [comment]
@iamgoofball just to know what you think about this.

**Comment by Iamgoofball** [comment]
dicks out for harambe

**Comment by nullbear** [comment]
y does this take so long 2 merge q_q

**Comment by optimumtact** [comment]
If this breaks chemistry horribly I'm going to delete you

**Comment by optimumtact** [comment]
So this broke master and was reverted
code/modules/reagents/chemistry/recipes/slime_extracts.dm:84:error: result: undefined var

code/modules/reagents/chemistry/recipes/slime_extracts.dm:86:error: result_amount: undefined var

**Comment by PKPenguin321** [comment]
ack

**Comment by nullbear** [comment]
bleh why are slimes silly.

**Comment by nullbear** [comment]
Ah, it's probably a merge conflict or something that isn't showing up, because the code runs fine on my end.

**Pull request merged**